test/DependencyInjection/BuilderExtensions/ResponsiveTests.cs

```diff
@@ -1,50 +1,148 @@
+        public void AddResponsive_Null_ArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(CreateResponsiveNullService);
+        }
+
+        [Fact]
+        public void AddResponsive_Options_Builder_Service()
+        {
+            var service  = new ServiceCollection();
+            var expected = service.Count + total;
+            var builder  = service.AddDetection(options => { options.Responsive.DefaultTablet = Device.Desktop; });
+
+            //Assert.Equal(expected, builder.Services.Count);
+            Assert.Same(service, builder.Services);
+        }
+
+        [Fact]
+        public void AddResponsive_Options_Null_ArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(CreateResponsiveNullService);
+        }
+
+        private readonly Func<object> CreateResponsiveNullService =
+            () => ((IServiceCollection) null).AddDetection();
+
+        [Fact]
+        public async void AddResponsive_Options_Disable_True()
+        {
+            var service  = MockService.CreateService("mobile");
+            var options  = new DetectionOptions {Responsive = {Disable = true}};
+            var resolver = MockResponsiveService(service, options);
+
+            var app = MockApplicationBuilder(options, resolver);
+
+            app.UseDetection();
+
+            var request = app.Build();
+
+            Assert.NotEqual(service.Context, new DefaultHttpContext());
+
+            await request.Invoke(service.Context);
+
+            Assert.Equal(Device.Mobile, service.Context.GetDevice());
+        }
+
+        [Fact]
+        public async void AddResponsive_Options_Disable_False()
+        {
+            var service  = MockService.CreateService("mobile");
+            var options  = new DetectionOptions {Responsive = {Disable = false}};
+            var resolver = MockResponsiveService(service, options);
+
+            var app = MockApplicationBuilder(options, resolver);
+
+            app.UseDetection();
+
+            var request = app.Build();
+
+            Assert.NotEqual(service.Context, new DefaultHttpContext());
+
+            await request.Invoke(service.Context);
+
+            Assert.Equal(Device.Mobile, service.Context.GetDevice());
+        }
+
+        [Fact]
+        public async void AddResponsive_Include_WebApi()
+        {
+            var service = MockService.CreateService("mobile");
+            service.Context.Request.Path = "/api/dog";
+            var options  = new DetectionOptions {Responsive = {IncludeWebApi = true}};
+            var resolver = MockResponsiveService(service, options);
+
+            var app = MockApplicationBuilder(options, resolver);
+
+            app.UseDetection();
+
+            var requestDelegate = app.Build();
+
+            Assert.NotEqual(service.Context, new DefaultHttpContext());
+
+            await requestDelegate.Invoke(service.Context);
+
+            Assert.Equal(Device.Mobile, service.Context.GetDevice());
```

Comment:
`GetDevice(this HttpConect context)` is returning `Device.Desktop` only. Maybe because it was never set in the unit test. Need a way to solve this in the function or the set it in the unit test.